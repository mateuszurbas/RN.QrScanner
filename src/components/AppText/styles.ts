import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Roboto_500Medium',
  },
});

export default styles;
