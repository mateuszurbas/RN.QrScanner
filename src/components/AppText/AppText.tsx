import React, { memo, ReactNode } from 'react';
import { Text, StyleProp, TextStyle } from 'react-native';

import styles from './styles';

type Props = {
  style?: StyleProp<TextStyle>;
  children: ReactNode;
};

const AppText = function ({ children, style }: Props) {
  return <Text style={[styles.text, style]}>{children}</Text>;
};

export default memo(AppText);
