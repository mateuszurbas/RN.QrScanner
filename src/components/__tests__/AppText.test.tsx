import * as React from 'react';
import renderer from 'react-test-renderer';

import { AppText } from '../AppText';

it(`Component AppText renders correctly`, () => {
  const tree = renderer.create(<AppText>Test</AppText>);
  expect(tree).toMatchSnapshot();
});
