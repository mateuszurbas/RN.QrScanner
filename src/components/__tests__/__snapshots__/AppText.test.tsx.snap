// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AppText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Roboto_500Medium",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
